tsk/fs/ntfs.c

```diff
@@ -5314,13 +5314,25 @@ ntfs_open(TSK_IMG_INFO * img_info, TSK_OFF_T offset,
         goto on_error;
     }
 
-    if ((ntfs->fs->csize != 0x01) &&
-        (ntfs->fs->csize != 0x02) &&
-        (ntfs->fs->csize != 0x04) &&
-        (ntfs->fs->csize != 0x08) &&
-        (ntfs->fs->csize != 0x10) &&
-        (ntfs->fs->csize != 0x20) && (ntfs->fs->csize != 0x40)
-        && (ntfs->fs->csize != 0x80)) {
+    uint32_t csize = ntfs->fs->csize;
+    if (ntfs->fs->csize > 0x80) {
+      csize = 1 << -(int8_t)ntfs->fs->csize;
+    }
+
+    if ((csize != 0x01) &&
+      (csize != 0x02) &&
+      (csize != 0x04) &&
+      (csize != 0x08) &&
+      (csize != 0x10) &&
+      (csize != 0x20) &&
+      (csize != 0x40) &&
+      (csize != 0x80) &&
+      (csize != 0x100) &&
+      (csize != 0x200) &&
+      (csize != 0x400) &&
+      (csize != 0x800) &&
+      (csize != 0x1000)
+      ) {
```

Comment:
Hi. I'm trying to clean up the code base a bit. Can you please add a comment explaining what's going on here? Thanks!